### PR TITLE
UBKit: Adds possibility to set query parameters as percent encoded query parameters

### PR DESCRIPTION
**Pull request by choli**

### CODE REVIEW

File: Sources/UBFoundation/Networking/UBURLRequest.swift [hunk 1]
```diff
@@ -188,41 +188,83 @@ public struct UBURLRequest: Equatable, Hashable, CustomReflectable, CustomString
...
+        try setQueryParameters(parameters.map { URLQueryItem(name: $0.key, value: $0.value) }, percentEncoded: true)
     }
 
     /// Sets the query parameter
     ///
     /// - Parameter parameter: A query item
     /// - Throws: `UBNetworkingError` in case of missing or malformed URL
-    public mutating func setQueryParameters(_ parameter: URLQueryItem) throws {
-        try setQueryParameters([parameter])
+    public mutating func setQueryParameter(_ parameter: URLQueryItem) throws {
```
Comment (stmitt) [inline Sources/UBFoundation/Networking/UBURLRequest.swift:210]: I would prefer to not change the method signature to not introduce a breaking change. (The unit tests are failing for that reason)

Comment (choli) [reply Sources/UBFoundation/Networking/UBURLRequest.swift:210]: Agreed. Would you keep the new method also with s or have it correctly spelled?

Comment (stmitt) [reply Sources/UBFoundation/Networking/UBURLRequest.swift:210]: I would keep the new one with the correct spelling. We could also deprecate the wrongly spelled one and add the correct one as a replacement but I don't think it's worth the hassle.

Comment (choli) [reply Sources/UBFoundation/Networking/UBURLRequest.swift:217]: Put it back in but deprecated, was my idea before as well but thought thats too much. Now it was easy to just copy paste it in there :)

**Review by stmitt — APPROVED** [review]

**Pull request merged**